src/app/components/sections/nested-challenges.tsx

```diff
@@ -478,45 +480,64 @@ const StagesContentSection = ({
   attemptsLoading,
   period,
   setPeriod,
+  challengeModules,
 }: {
   challenge: Course;
   attempts: ChallengeAttempt[];
   attemptsLoading: boolean;
   period: ChallengePeriod;
   setPeriod: React.Dispatch<React.SetStateAction<ChallengePeriod>>;
+  challengeModules: Course[];
 }) => {
+  const { allRepositories } = useStore();
+  const repository = allRepositories.find(
+    (repo) => sluggify(repo?.challenge?.title) === sluggify(challenge.title)
+  );
+
+  const isCompleted = repository?.progress?.status === Status.Completed;
```

Comment:
The Status.Completed in types folder is not the same as the one for progress status.

```suggestion
  const isCompleted = repository?.progress?.status === "completed";
```